django/forms/boundfield.py

```diff
@@ -219,12 +219,28 @@ def initial(self):
         return data
 
     def build_widget_attrs(self, attrs, widget=None):
-        widget = widget or self.field.widget
         attrs = dict(attrs)  # Copy attrs to avoid modifying the argument.
-        if widget.use_required_attribute(self.initial) and self.field.required and self.form.use_required_attribute:
-            attrs['required'] = True
+        if isinstance(widget, MultiWidget) or isinstance(self.field.widget, MultiWidget):
+            widgets = widget.widgets or self.field.widget.widgets
```

Comment:
I think leave the original `widget = widget or self.field.widget` here. Then do the `isinstance()` check, so that we're no repeating ourselves in the `else` block. 

`widgets` might then be clearer as `subwidgets`. (Or such.)